File: src/main/ruby/advent2023/PuzzleXyz.rb
load 'base.rb'

module PuzzleXyz
  class Puzzle
    include Base

    def main
      solution1 = self.solve_puzzle("puzzleXYZ/inputs.txt", Array.new) do |line, data|
        parse_line(data, line)
      end
      puts "Solution1: #{solution1}"

      solution2 = self.solve_puzzle2("puzzleXYZ/inputs.txt", Array.new) do |line, data|
        parse_line(data, line)
      end
      puts "Solution2: #{solution2}"
    end

    def parse_line(data, line)
      data << line.to_i
    end

    def compute_solution(data)
      data.sum
    end

    def compute_solution2(data)
      data.sum + 5
    end
  end
end

p = PuzzleXyz::Puzzle.new
p.main

File: src/main/python/advent2024/PuzzleXYZ.py
from Base import Base
from typing import Tuple

MyDataType = list[int]

class Puzzle(Base[MyDataType, int, int]):
    def main(self):
      solution1: int = self.solve_puzzle("puzzleXYZ/inputs.txt", MyDataType())  # type: ignore
      print("Solution1: {}".format(solution1))

      solution2: int = self.solve_puzzle2("puzzleXYZ/inputs.txt", MyDataType()) # type: ignore
      print("Solution2: {}".format(solution2))

    def read_one_line(self, line: str, data: MyDataType):
      data.append(int(line))

    def compute_solution(self, data: MyDataType) -> int:
      pass

    def compute_solution2(self, data: MyDataType):
      pass
    
puzzle: Puzzle = Puzzle()
puzzle.main()